test/specs/commonTests/implementsShorthandProp.js

```diff
@@ -32,9 +33,11 @@ export default (Component, options = {}) => {
     ShorthandComponent,
     shorthandDefaultProps = {},
     shorthandOverrideProps = {},
+    strictAssert = true,
     requiredProps = {},
   } = options
   const { assertRequired } = helpers('implementsShorthandProp', Component)
+  const assertMethod = strictAssert ? 'contain' : 'containMatchingElement'
```

Comment:
Why? [`contains`](https://github.com/airbnb/enzyme/blob/master/docs/api/ShallowWrapper/contains.md) is too strict while I added `onClick` handler to the `Icon` it was real pain. I think it's a quite elegant solution, as I remember we had same problem with the `Input` component.